src/kernels/raw/finder/pythonKernelInterruptDaemon.node.ts

```diff
@@ -3,51 +3,171 @@
 
 'use strict';
 
-import { ChildProcess } from 'child_process';
-import { MessageConnection, RequestType0 } from 'vscode-jsonrpc';
-import { traceInfo } from '../../../platform/logging';
-import { IPlatformService } from '../../../platform/common/platform/types';
-import { BasePythonDaemon } from '../../../platform/common/process/baseDaemon.node';
-import { IPythonExecutionService } from '../../../platform/common/process/types.node';
-import { PythonEnvironment } from '../../../platform/pythonEnvironments/info';
-import { IPythonKernelDaemon } from '../types';
+import { traceError, traceVerbose, traceWarning } from '../../../platform/logging';
+import { IPythonExecutionFactory, ObservableExecutionResult } from '../../../platform/common/process/types.node';
+import { EnvironmentType, PythonEnvironment } from '../../../platform/pythonEnvironments/info';
+import { inject, injectable } from 'inversify';
+import { IInterpreterService } from '../../../platform/interpreter/contracts';
+import { IAsyncDisposable, IDisposableRegistry, IExtensionContext, Resource } from '../../../platform/common/types';
+import { createDeferred, Deferred } from '../../../platform/common/utils/async';
+import { Disposable, Uri } from 'vscode';
+import { EOL } from 'os';
+function isBestPythonInterpreterForAnInterruptDaemon(interpreter: PythonEnvironment) {
+    if (
+        isSupportedPythonVersion(interpreter) &&
+        (interpreter?.envType === EnvironmentType.Global ||
```

Comment:
Maybe a comment? I'm not sure why the envType is restricted.

Reply:
Done